alpha/apps/kaltura/modules/extwidget/actions/serveFlavorAction.class.php

```diff
@@ -799,4 +798,84 @@ private function getClipData($path, $flavor)
 		return $clipDesc ;
 	}
 
-}
\ No newline at end of file
+	protected function serveLiveChannel(LiveChannel $entry)
+	{
+		$repeat = $entry->getRepeat();
+		$playlist = entryPeer::retrieveByPK($entry->getPlaylistId());
+
+		list($entryIds, $durations, $referenceEntry, $captionFiles) = myPlaylistUtils::executeStitchedPlaylist($playlist, null);
+
+		// get cycle info
```

Comment:
Try to break this long function into smaller pieces - very hard to read and follow